apps/easypid/src/features/didcomm/useDidCommPresentationNotificationSlides.tsx

```diff
@@ -0,0 +1,104 @@
+import { useAgent, useDidCommPresentationActions } from '@package/agent'
+import { useToastController } from '@package/ui'
+import React from 'react'
+
+import type { SlideStep } from '@package/app/src'
+import { addSharedActivityForSubmission } from '../activity/activityRecord'
+import { PresentationSuccessSlide } from '../share/slides/PresentationSuccessSlide'
+import { ShareCredentialsSlide } from '../share/slides/ShareCredentialsSlide'
+
+interface DidCommPresentationNotificationSlidesProps {
+  proofExchangeId: string
+  onCancel: () => void
+  onComplete: () => void
+}
+
+export function useDidCommPresentationNotificationSlides({
+  proofExchangeId,
+  onCancel,
+  onComplete,
+}: DidCommPresentationNotificationSlidesProps) {
+  const { agent } = useAgent()
+  const toast = useToastController()
+  const { acceptPresentation, declinePresentation, proofExchange, acceptStatus, submission, verifierName } =
+    useDidCommPresentationActions(proofExchangeId)
+
+  const onProofAccept = async () => {
+    if (!submission) return
+
+    await acceptPresentation({})
+      .then(async () => {
+        await addSharedActivityForSubmission(
+          agent,
+          submission,
+          {
+            id: proofExchangeId,
+            name: verifierName,
+          },
+          'success'
+        )
+      })
+      .catch(async () => {
```

Comment:
small chance, but if `addSharedActivityForSubmission` fails after successfull sharing, it will then call `addSharedActivityForSubmission` in catch again, and show the toast `Presentation could not be shared.`.